apps/hash-graph/lib/graph/src/snapshot/error.rs

```diff
@@ -0,0 +1,35 @@
+use std::{error::Error, fmt};
+
+#[derive(Debug)]
+pub enum SnapshotDumpError {
+    Read,
+    Query,
+    Write,
+}
+
+impl fmt::Display for SnapshotDumpError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Read => write!(f, "could not read a snapshot record"),
+            Self::Query => write!(f, "could not query snapshot records from the store"),
+            Self::Write => write!(f, "could not write a snapshot record into the sink"),
+        }
+    }
+}
+
+impl Error for SnapshotDumpError {}
+
+#[derive(Debug)]
+pub enum SnapshotRestoreError {
+    Unsupported,
+}
+
+impl fmt::Display for SnapshotRestoreError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Unsupported => write!(f, "The snapshot contains unsupported entries"),
```

Comment:
Do we log the entries that are unsupported? Or does this just happen whenever there's a `serde` error (in which case I guess we don't know if it's unsupported or malformed)

Reply:
Currently, any value is unsupported. The logic is not implemented at all and I just did some outputs to the log to test the stream interface generally. Every call to `restore_snapshot` currently will `panic`.